### DISCUSSION TITLE
Adafruit_USBD_MSC' does not name a type

### DISCUSSION THREAD

**Discussion by Bert-coder** [adafruit/Adafruit_TinyUSB_Arduino#483] [Q&A]
Hello,
I have used: https://github.com/adafruit/Adafruit_TinyUSB_Arduino/blob/master/examples/MassStorage/msc_esp32_file_browser/msc_esp32_file_browser.ino.
I followed the instructions and updated all libraries. I keep getting errors: "Adafruit_USBD_MSC' does not name a type"
The boards I have tried: MH-ET LIVE ESP32 minikit and  MH-ET LIVE ESP32 devkit.
the error message:
usb_hub_23-12-24:68:1: error: 'Adafruit_USBD_MSC' does not name a type; did you mean 'Adafruit_USBH_CDC'?
   68 | Adafruit_USBD_MSC usb_msc;
      | ^~~~~~~~~~~~~~~~~
      | Adafruit_USBH_CDC
C:\Users\bram\OneDrive\Documents\Arduino\usb_hub_23-12-24\usb_hub_23-12-24.ino: In function 'void setupMassStorage()':
usb_hub_23-12-24:87:3: error: 'usb_msc' was not declared in this scope
   87 |   usb_msc.setID("Adafruit", "External Flash", "1.0");
      |   ^~~~~~~
usb_hub_23-12-24:101:7: error: 'TinyUSBDevice' was not declared in this scope
  101 |   if (TinyUSBDevice.mounted()) {
      |       ^~~~~~~~~~~~~
Meerdere bibliotheken gevonden voor "WiFi.h"
Gebruikt: C:\Users\bram\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.1.0\libraries\WiFi
Niet gebruikt: C:\Program Files (x86)\Arduino\libraries\WiFi
exit status 1
'Adafruit_USBD_MSC' does not name a type; did you mean 'Adafruit_USBH_CDC'?

Thanks in advance

**Comment by hathach** [comment]
what is your IDE menu setting, you will need to enable CDC, but leave MSC and DFU as disabled.
